mftf/2.3/reporting.md

```diff
@@ -0,0 +1,367 @@
+
+```terminal
+AdminLoginTestCest: Admin login test
+Signature: Magento\AcceptanceTest\_default\Backend\AdminLoginTestCest:AdminLoginTest
+Test: tests/functional/Magento/FunctionalTest/_generated/default/AdminLoginTestCest.php:AdminLoginTest
+Scenario --
+ I am on page "/admin/admin"
+ I fill field "#username","admin"
+ I fill field "#login","123123q"
+ I click ".actions .action-primary"
+ I wait for page load 30
+ I close admin notification
+ I see in current url "/admin/admin"
+ PASSED
+```
+{: .no-copy }
+
+The running test is `AdminLoginTestCest`, which is `Admin login test` (this text is generated from the test name but with the `Cest` part excluded).
+Its test signature is `Magento\AcceptanceTest\_default\Backend\AdminLoginTestCest:AdminLoginTest` that matches a `className:methodName` format using namespaces.
+A path to the corresponding `Test` is `tests/functional/Magento/FunctionalTest/_generated/default/AdminLoginTestCest.php:AdminLoginTest` (relative to the `acceptance/` directory).
+
+`Scenario` lists the tests steps as they run during test execution, ending with the successful test verdict `PASSED`.
+It means that all test steps were processed as expected.
+
+#### Failed tests
+
+The second test fails with the following report:
+
+```terminal
+AdminMenuNavigationWithSecretKeysTestCest: Admin menu navigation with secret keys test
+Signature: Magento\AcceptanceTest\_default\Backend\AdminMenuNavigationWithSecretKeysTestCest:AdminMenuNavigationWithSecretKeysTest
+Test: tests/functional/Magento/FunctionalTest/_generated/default/AdminMenuNavigationWithSecretKeysTestCest.php:AdminMenuNavigationWithSecretKeysTest
+Scenario --
+ I magento cli "config:set admin/security/use_form_key 1"
+ Value was saved.
+ I magento cli "cache:clean config full_page"
+ Cleaned cache types:
+config
+full_page
+ I am on page "/admin/admin"
+ I wait for page load
+ I fill field "#username","admin"
+ I fill field "#login","123123q"
+ I click ".actions .action-primary"
+ I wait for page load 30
+ I close admin notification
+ I click "//li[@id='menu-magento-backend-stores']"
+ I wait for loading mask to disappear
+ I click "#nav li[data-ui-id='menu-magento-config-system-config']"
+ I wait for page load
+ I see current url matches "~\/admin\/system_config\/~"
+ I see "#something"
+ I save screenshot
+ FAIL
+
+ I magento cli "config:set admin/security/use_form_key 0"
+ Value was saved.
+ I magento cli "cache:clean config full_page"
+ Cleaned cache types:
+config
+full_page
+ I am on page "/admin/admin/auth/logout/"
+--------------------------------------------------------------------------------
+```
+{: .no-copy }
+
+The general test details and scenario has the same format as in the Passed test.
+The interesting part starts near the `FAIL` line.
+
+```terminal
+ I see "#something"
+ I save screenshot
+ FAIL
+```
+
+When a test step fails, the MFTF always saves a screenshot of the web page with failing state.
```

Comment:
... with the failing state in html and .png format to the `acceptance/tests/_output` directory ....

Reply:
It will introduce too much new semantical units into one sentence. It's like having a function with multiple parameters, the more pararmeters the function depends on - the more complex it to keep in mind as a whole concept.